src/ansys/fluent/core/utils/search.py

```diff
@@ -212,10 +251,230 @@ def inner(tree, path, root_path):
+            return correct_spell
+        elif match_case:
+            corrections_in_tree = set()
+            for correction in possible_corrections:
+                for name in names:
+                    if correction in name:
+                        corrections_in_tree.add(correction)
+            return list(corrections_in_tree)
+    else:
+        return [word]
+
+
+def search(
+    search_string: str,
+    language: Optional[str] = "eng",
+    wildcard: Optional[bool] = False,
+    match_whole_word: Optional[bool] = False,
+    match_case: Optional[bool] = True,
+):
+    """Search for a word through the Fluent's object hierarchy.
+
+    Parameters
+    ----------
+    search_string: str
+        The word to search for. Semantic search is default.
+    language: str
+        The language for the semantic search.
+        English is default for the semantic search.
+        ISO 639-3 code of the language to be used for semantic search.
+        See `https://omwn.org/omw1.html` for the list of supported languages.
+        The default value is `eng` for English language.
+    wildcard: bool
+        Whether to use wildcard pattern. If ``True`` will match wildcard pattern based on ``fnmatch`` module and
+        will turn off semantic matching.
+    match_whole_word: bool
+        Whether to get exact match. If ``True`` will match exact string and will turn off semantic matching.
+    match_case: bool
+        Whether to match case. If ``True`` will match case-insensitive case.
+
+    Raises
+    ------
+    ValueConflict
+        If both ``wildcard`` and ``match_whole_word`` are ``True``.
+
+    Examples
+    --------
+    >>> import ansys.fluent.core as pyfluent
+    >>> pyfluent.search("font", match_whole_word=True)
+    >>> pyfluent.search("Font")
+    >>> pyfluent.search("iter*", wildcard=True)
+    >>> pyfluent.search("读", language="cmn")   # search 'read' in Chinese
+    The most similar API objects are:
+    <solver_session>.file.read (Command)
+    <solver_session>.file.import_.read (Command)
+    <solver_session>.mesh.surface_mesh.read (Command)
+    <solver_session>.tui.display.display_states.read (Command)
+    <meshing_session>.tui.display.display_states.read (Command)
+    """
+    if wildcard and match_whole_word:
+        raise ValueConflict()
+    elif language and match_whole_word:
+        warnings.warn(
+            "``match_whole_word=True`` will match exact string and will turn off semantic matching.",
+            UserWarning,
+        )
+    elif match_whole_word:
+        warnings.warn(
+            "``match_whole_word=True`` will turn off semantic and wildcard matching.",
+            UserWarning,
+        )
+
+    api_object_names = get_api_tree_file_name(name=True)
+    names_txt = open(api_object_names, "r")
+    names = [name.replace("\n", "") for name in names_txt.readlines()]
+    synset_1 = wn.synsets(search_string, lang=language)
+
+    if wildcard and not match_whole_word:
```

Comment:
if wildcard then not match_whole_word since wildcard and not match_whole_word is not allowed so wildcard and not match_whole_word is wildcard and wildcard.

Reply:
Right, I added exception later and this condition should have updated according to it. Thank you.